scheduler/src/cook/mesos/api.clj

```diff
@@ -200,11 +203,34 @@
   [s]
   (re-matches #"[\.a-zA-Z0-9_-]{1,128}" s))
 
+(defn non-empty-max-128-characters
+  [s]
+  (< 0 (.length s) 129))
+
+(defn date?
+  [s]
+  (try
+    (tf/parse partition-date-format s)
+    true
+    (catch Exception e false)))
+
 (def Application
   "Schema for the application a job corresponds to"
   {:name (s/constrained s/Str non-empty-max-128-characters-and-alphanum?)
    :version (s/constrained s/Str non-empty-max-128-characters-and-alphanum?)})
 
+(def Dataset
+  "Schema for a job dataset"
+  {:dataset {(s/constrained s/Str non-empty-max-128-characters)
+             (s/constrained s/Str non-empty-max-128-characters)}
```

Comment:
Can we please place the key and value on the same line?

Reply:
Can we please extract `(s/constrained s/Str non-empty-max-128-characters)` into a `def`?